enzy_htp/_interface/amber_interface.py

```diff
@@ -2321,6 +2344,15 @@ def get_coordinates(self, mdcrd: str) -> Generator[List[List[float]], None, None
         line_feed = os.linesep
         digit_pattern = r'[ ,\-,0-9][ ,\-,0-9][ ,\-,0-9][0-9]\.[0-9][0-9][0-9]' # a number
         frame_sep_pattern = digit_pattern * 3 + line_feed
+
+        # remove solvent
+        if remove_solvent:
+            temp_dir = eh_config["system.SCRATCH_DIR"]
+            fs.safe_mkdir(temp_dir)
+            new_mdcrd = fs.get_valid_temp_name(f"{temp_dir}/mdcrd_parser_temp_rm_sol.mdcrd")
+            self.parent_interface.remove_traj_solvent(mdcrd, self.prmtop_file, new_mdcrd)
+            mdcrd = new_mdcrd
+
         with open(mdcrd) as f:
```

Comment:
You can do this without having to use a while statement and it would be simpler I believe. you basically just want to parse in the whole thing as a giant buffer and read through it